### DISCUSSION TITLE
use 'document' in custom hook

### DISCUSSION THREAD

**Discussion by umerjaved178** [vercel/next.js#35672] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

typical environment

### What browser are you using? (if relevant)

Chrome

### Describe the Bug

We can't use 'document' in custom hook as 'document' doesn't exist on server side. Dynamic import only works with components.
How to handle this situation 

### Expected Behavior

There must be a way to use 'document' with next js

### To Reproduce

just make a simple custom hook that involves 'document', import it to next js page and error 
PS: Dynamic imports only work in case of Components not for hooks

**Comment by ijjk** [comment]
Hi, you can achieve this by leveraging `useEffect()` inside of the custom hook since `useEffect()` is not called on the server.
